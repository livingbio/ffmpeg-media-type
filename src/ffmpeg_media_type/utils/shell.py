import os
import shlex
import tempfile

from ..exceptions import FfmpegMediaTypeError


def call(cmds: list[str]) -> str:
    # call command and return stdout

    command = " ".join(shlex.quote(part) for part in cmds)
    with tempfile.TemporaryDirectory() as tmpdir:
        output_path = f"{tmpdir}/output.txt"
        if os.system(f"{command} > {output_path}") != 0:
            raise FfmpegMediaTypeError(f"command failed: {command}")

        with open(output_path) as ifile:
            return ifile.read()
