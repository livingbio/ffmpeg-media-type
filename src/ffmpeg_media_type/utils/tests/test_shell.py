from syrupy.assertion import SnapshotAssertion

from ..shell import call


def test_call(snapshot: SnapshotAssertion) -> None:
    assert snapshot == call(["echo", "hello world"])
