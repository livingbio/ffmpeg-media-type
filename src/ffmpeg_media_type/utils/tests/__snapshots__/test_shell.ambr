# serializer version: 1
# name: test_call
  '''
  hello world
  
  '''
# ---
